sunpy/visualization/imageanimator.py

```diff
@@ -608,32 +585,257 @@ def _sanitize_axis_range(self, axis_range, data):
+        for i, d in enumerate(data.shape):
+            if len(axis_range[i]) == d:
+                if i not in self.slider_axes:
+                    raise ValueError("Slider axes mis-match, non-slider axes need [min,max] pairs")
+        return axis_range
+
+
+
+class LineAnimator(ArrayAnimator):
+    """Create a matplotlib backend independent data explorer for 1D plots.
+
+    The following keyboard shortcuts are defined in the viewer:
+
+    - 'left': previous step on active slider
+    - 'right': next step on active slider
+    - 'top': change the active slider up one
+    - 'bottom': change the active slider down one
+    - 'p': play/pause active slider
+
+    This viewer can have user defined buttons added by specifying the labels
+    and functions called when those buttons are clicked as keyword arguments.
+
+    Parameters
+    ----------
+    data: ndarray
+        The y-axis data to be visualised = 2D
+
+    data_yaxis: `int`
+        The axis used to plot against xdata.
+
+    fig: `matplotlib.figure`
+        Figure to use
+
+    axis_range: `list` of physical coordinates for array or None
+        If None array indices will be used for all axes.
+        X-axis values must be supplied (if desired) as an array in the
+        element of axis_range corresponding to the data_yaxis in the
+        data input arg, i.e. xdata = axis_range[data_yaxis].
+        Also, the number of x-axis values must correspond to the number
+        of y-axis values, i.e. len(axis_range[data_yaxis]) must equal
+        len(data[data_yaxis]).
+        For the slider axes a [min, max] pair can be specified or an array the
+        same length as the axis which will provide all values for that slider.
+        If None is specified for an axis then the array indices will be used
+        for that axis.
+
+    xlabel: `str`
+        Label of x-axis of plot.
+
+    ylabel: `str`
+        Label of y-axis of plot.
+
+    xlim: `tuple`
+        Limits of x-axis of plot.
+
+    ylim: `tuple`
+        Limits of y-axis of plot.
+
+    interval: `int`
+        Animation interval in ms
+
+    button_labels: `list`
+        List of strings to label buttons
+
+    button_func: `list`
+        List of functions to map to the buttons
+
+    Extra keywords are passed to plot.
+
+    """
+
+    def __init__(self, data, data_yaxis=0, axis_range=None, ylabel=None, xlabel=None,
+                 xlim=None, ylim=None, **kwargs):
+        # Check inputs.
+        data_yaxis = int(data_yaxis)
+        if data_yaxis != 0 and data_yaxis != 1:
+            raise ValueError("data_yaxis must be either 0 or 1 referring to "
+                             "the axis of data to be used for a single plot.")
+        if data.ndim != 2:
+            raise ValueError("data must have two dimensions.  One for data "
+                             "for each single plot and one for time/iteration.")
+        # Attach data to class.
+        axis_range[data_yaxis] = np.array(axis_range[data_yaxis])
```

Comment:
You're right.  It's defined in the ArrayAnimator init so this has to be after that.